.devops/nix/package.nix

```diff
@@ -157,6 +157,11 @@ effectiveStdenv.mkDerivation (
       substituteInPlace ./*.py --replace "/usr/bin/env python" "${llama-python}/bin/python"
     '';
 
+    # requires sandboxing to be off or relaxed on MacOS (off by default)
+    preConfigure = lib.optionals useMetalKit ''
+      export PATH=/usr/bin:$PATH # make sure /usr/bin/xcrun is on PATH
+    '';
```

Comment:
I'm not particularly familiar with Darwin and I don't know what xcrun is. Could you attach a brief comment expounding this and explaining why it's needed for the `configurePhase`? What else is exposed in `/usr/bin` with the relaxed sandbox? Why do you give `/usr/bin` a higher priority than to the nix-provided `PATH`? If `xcrun` is the only thing you need from `/usr/bin` you could instead make a wrapper for it:

```
nativeBuildInputs = optionals hostPlatform.isDarwin [
  (writeShellScriptBin "xcrun" "/usr/bin/scrun")
];
```

A more general darwin question: can we explicitly request the relaxed sandbox using some kind of `requiredSystemFeatures`?

Reply:
It should not be necessary to taint the build with `/usr/bin/xcrun` if all we need is `xcrun` -- it can be found in the `xcbuild` and `xcodebuild` packages.

Reply:
@swdunlop not having any progress with `xcbuild` package -- I wonder if it's too old to know about the `metal` shader compiler. Looks like its Github project has been archived since mid-2021. 

When attempting to run `nix run nixpkgs#xcbuild` and `nix run nixpkgs#xcodebuild`, get the following error:
```
warning: non-build action metal not implemented
error: unknown build action 'metal
```

Reply:
@SomeoneSerge `xcrun` is a MacOS/Xcode tool that finds other command line tools--for example, finding the `metal` shader compiler. I think it's used instead of a hard-coded path because different installations of Xcode can coexist, so `xcrun` will run whichever tool has been made the default by `xcode-select`.

Reply:
Man page for `xcrun` and `Xcode-select` [here](https://www.manpagez.com/man/1/xcode-select/)